components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java

```diff
@@ -865,16 +876,31 @@ protected boolean canUserBeProvisioned(ProvisioningEntity provisioningEntity,
             // we apply restrictions only for users.
             // if service provider's out-bound provisioning configuration does not define any roles
             // to be provisioned then we apply no restrictions.
+            /*  After group role separation, the internal roles should be not be provisioned. Only groups should be
+                provisioned.
+            */
+            if (isInternalRole(provisioningEntity.getEntityName(), tenantDomain)) {
```

Comment:
Add a comment saying the entity name can be group name or role-uuid depending on group or role assignment for a user. Hence need to check whether group or role to only allow provisioning groups.

Reply:
Better way to stop role get provisioned.